api/contract.go

```diff
@@ -181,13 +193,24 @@ func (s *ContractService) CallMethod(r *http.Request, args *CallMethodArgs, re *
 		return errors.Wrap(err, "can't get objectRef")
 	}
 
+	pulse, err := s.runner.PulseAccessor.Latest(ctx)
+	if err != nil {
+		return errors.Wrap(err, "can't get current pulse")
+	}
+	hash := sha256.New()
+	_, err = hash.Write(args.MethodArgs)
+	if err != nil {
+		return errors.Wrap(err, "Cant get hash")
+	}
+
 	msg := &message.CallMethod{
 		IncomingRequest: record.IncomingRequest{
 			Caller:       testutils.RandomRef(),
 			Object:       objectRef,
 			Method:       args.Method,
 			Arguments:    args.MethodArgs,
 			APIRequestID: utils.TraceID(ctx),
+			Reason:       *insolar.NewReference(*insolar.NewID(pulse.PulseNumber, hash.Sum(nil))),
```

Comment:
can we make special method for generate reason? I see copy-paste code, it can produce bug's later, when you change code in one place and doesn't change in another.

Reply:
And i'm sure, that reason generates not only in this two places

Reply:
done